tests/appsec/test_automated_login_events.py

```diff
@@ -1348,3 +1218,761 @@ def validate_iden(meta):
+            assert meta["appsec.events.users.login.failure.track"] == "true"
+            assert meta["_dd.appsec.events.users.login.failure.sdk"] == "true"
+            assert meta["appsec.events.users.login.failure.usr.exists"] == "true"
+
+    def setup_login_sdk_failure_local(self):
+        self.r_sdk_failure = weblog.post(
+            "/login?auth=local&sdk_event=failure&sdk_user=sdkUser&sdk_user_exists=true",
+            data=login_data(context, INVALID_USER, PASSWORD),
+        )
+
+    def test_login_sdk_failure_local(self):
+        assert self.r_sdk_failure.status_code == 401
+        for _, trace, span in interfaces.library.get_spans(request=self.r_sdk_failure):
+            assert_priority(span, trace)
+            meta = span.get("meta", {})
+
+            # mandatory
+            assert meta["appsec.events.users.login.failure.usr.login"] == "sdkUser"
+            assert meta["_dd.appsec.usr.login"] == INVALID_USER_HASH
+            assert meta["_dd.appsec.events.users.login.failure.auto.mode"] == "anonymization"
+            assert meta["appsec.events.users.login.failure.track"] == "true"
+            assert meta["_dd.appsec.events.users.login.failure.sdk"] == "true"
+            assert meta["appsec.events.users.login.failure.usr.exists"] == "true"
+
+    def setup_signup_local(self):
+        self.r_success = weblog.post("/signup", data=login_data(context, NEW_USER, PASSWORD))
+
+    @missing_feature(context.library == "nodejs", reason="Signup events not implemented")
+    def test_signup_local(self):
+        assert self.r_success.status_code == 200
+        for _, trace, span in interfaces.library.get_spans(request=self.r_success):
+            assert_priority(span, trace)
+            meta = span.get("meta", {})
+
+            # mandatory
+            assert meta["appsec.events.users.signup.usr.login"] == NEW_USERNAME_HASH
+            assert meta["_dd.appsec.usr.login"] == NEW_USERNAME_HASH
+            assert meta["_dd.appsec.events.users.signup.auto.mode"] == "anonymization"
+            assert meta["appsec.events.users.signup.track"] == "true"
+
+            # optional (to review for each library)
+            if context.library not in libs_without_user_id:
+                assert meta["appsec.events.users.signup.usr.id"] == NEW_USER_HASH
+                assert meta["_dd.appsec.usr.id"] == NEW_USER_HASH
+
+
+DISABLED = ("datadog/2/ASM_FEATURES/auto-user-instrum/config", {"auto_user_instrum": {"mode": "disabled"}})
+IDENTIFICATION = ("datadog/2/ASM_FEATURES/auto-user-instrum/config", {"auto_user_instrum": {"mode": "identification"}})
+ANONYMIZATION = ("datadog/2/ASM_FEATURES/auto-user-instrum/config", {"auto_user_instrum": {"mode": "anonymization"}})
+
+
+@rfc("https://docs.google.com/document/d/1RT38U6dTTcB-8muiYV4-aVDCsT_XrliyakjtAPyjUpw")
+@features.user_monitoring
+@scenarios.appsec_auto_events_rc
+class Test_V3_Login_Events_RC:
```

Comment:
are you checking that the capability is sent by the library ? i can't see it

Reply:
Yep it's possible and I forgot, but I've added it, thanks for the input! :smile: